src/utils.ts

```diff
@@ -166,3 +172,37 @@ export function getProvider(chainId: ChainId): ethers.providers.BaseProvider {
 
   return ethers.getDefaultProvider(ethers.providers.getNetwork(chainId));
 }
+
+export async function validateAmount(amount: number, maxAvalible: number): Promise<void> {
+  if (amount <= 0) {
+    return Promise.reject(new Error('Amount should be greater than 0'));
+  }
+  if (amount > maxAvalible) {
+    return Promise.reject(new Error('Amount cant be greater than total avalible Magic Tokens'));
+  }
+}
+
+export async function contractApprove(
```

Comment:
again, `src/utils` is a generic file. why would you assume that something called `contractApprove` would only work with Magic if it's located here? I recommend that you move these functions to a dedicated hook. See how it's done in https://github.com/chronologic/automate-ui/tree/main/src/components/BatchTx